db/memtable.cc

```diff
@@ -560,16 +560,28 @@ FragmentedRangeTombstoneIterator* MemTable::NewRangeTombstoneIteratorInternal(
         read_seq);
   }
 
-  auto* unfragmented_iter = new MemTableIterator(
-      *this, read_options, nullptr /* arena */, true /* use_range_del_table */);
-  auto fragmented_tombstone_list =
-      std::make_shared<FragmentedRangeTombstoneList>(
-          std::unique_ptr<InternalIterator>(unfragmented_iter),
-          comparator_.comparator);
-
-  auto* fragmented_iter = new FragmentedRangeTombstoneIterator(
-      fragmented_tombstone_list, comparator_.comparator, read_seq);
-  return fragmented_iter;
+  // takes current cache
+  std::shared_ptr<std::shared_ptr<FragmentedRangeTombstoneListCache>> cache =
+      std::atomic_load_explicit(cached_range_tombstone_.Access(),
```

Comment:
I don't know what locking this may do internally, but it's fine with me since this does not run when there's zero range tombstones.

Reply:
I don't know either. This should be the same as .load(std::memory_order_relaxed) for other atomic types, but shared_ptr requires this syntax.